ircd/s_newconf.c

```diff
@@ -685,23 +685,39 @@ time_t
 valid_temp_time(const char *p)
 {
 	time_t result = 0;
-
-	while(*p)
-	{
-		if(IsDigit(*p))
-		{
-			result *= 10;
-			result += ((*p) & 0xF);
-			p++;
-		}
-		else
+	int current = 0;
+	char *ch = (char *)p, *och;
+
+	while (*ch) {
+		och = ch;
+		current = strtol(ch, &ch, 10);
```

Comment:
Please check for range errors (return value of LONG_MIN/LONG_MAX + errno == ERANGE)